@xen-orchestra/web-core/lib/components/ui/input/UiInput.vue

```diff
@@ -1,141 +1,172 @@
-<!-- WIP -->
+<!-- v5 -->
 <template>
-  <div class="ui-input">
-    <VtsIcon :icon accent="current" class="before" />
-    <input :id v-model.trim="modelValue" class="typo p1-regular input" type="search" v-bind="attrs" />
-    <VtsIcon
-      v-if="!attrs.disabled && modelValue"
-      :icon="faXmark"
-      class="after"
-      accent="info"
-      @click="modelValue = ''"
-    />
+  <div class="ui-input" :class="toVariants({ accent })">
+    <UiLabel v-if="slots.default" :accent="labelAccent" :required :icon :href>
+      <slot />
+    </UiLabel>
+    <div>
+      <VtsIcon :icon="iconBefore" accent="current" class="before" />
+      <input v-model.trim="modelValue" class="typo p1-regular input text-truncate" :type :disabled v-bind="attrs" />
+      <VtsIcon
+        v-if="!attrs.disabled && modelValue && clearable"
+        :icon="faXmark"
+        class="after"
+        accent="info"
+        @click="modelValue = ''"
+      />
+    </div>
+    <UiInfo v-if="slots.info" :accent>
+      <slot name="info" />
+    </UiInfo>
   </div>
 </template>
 
 <script lang="ts" setup>
 import VtsIcon from '@core/components/icon/VtsIcon.vue'
-import { uniqueId } from '@core/utils/unique-id.util'
+import UiInfo from '@core/components/ui/info/UiInfo.vue'
+import UiLabel from '@core/components/ui/label/UiLabel.vue'
+import { toVariants } from '@core/utils/to-variants.util'
 import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
 import { faXmark } from '@fortawesome/free-solid-svg-icons'
 import { computed, useAttrs } from 'vue'
 
+type InputTextAccent = 'info' | 'warning' | 'danger'
+type InputType = 'text' | 'number' | 'password'
+
 defineOptions({
   inheritAttrs: false,
 })
-
-defineProps<{
+const props = defineProps<{
+  type: InputType
+  accent: InputTextAccent
+  disabled?: boolean
+  required?: boolean
+  clearable?: boolean
+  href?: string
   icon?: IconDefinition
+  iconBefore?: IconDefinition
 }>()
```

Comment:
Since we’re using Vue 3.5, you can use [props destructuring](https://vuejs.org/guide/components/props.html#reactive-props-destructure) here:
```suggestion
const { accent } = defineProps<{
  type: InputType
  accent: InputTextAccent
  disabled?: boolean
  required?: boolean
  clearable?: boolean
  href?: string
  icon?: IconDefinition
  iconBefore?: IconDefinition
}>()
```
I think it would add a little more clarity on which _prop_ is used in the script.

Reply:
Very nice